src/Interpreters/Cache/Metadata.h

```diff
@@ -107,6 +108,7 @@ struct CacheMetadata : public std::unordered_map<FileCacheKey, KeyMetadataPtr>,
     static String getFileNameForFileSegment(size_t offset, FileSegmentKind segment_kind);
 
     void iterate(IterateFunc && func);
+    bool isEmpty() const;
```

Comment:
this overload can also be named `empty()`, just so it won't stand out from the crowd